src/client/opamClient.ml

```diff
@@ -635,12 +635,118 @@ let init_checks ?(hard_fail_exn=true) init_config =
   if hard_fail && hard_fail_exn then OpamStd.Sys.exit_because `Configuration_error
   else not (soft_fail || hard_fail)
 
-let windows_checks ?cygwin_setup config =
+let git_for_windows_check =
+  if not Sys.win32 && not Sys.cygwin then fun ?gitbin:_ () -> None else
+  fun ?gitbin () ->
+    let header () = OpamConsole.header_msg "Git" in
+    let contains_git p =
+      OpamSystem.resolve_command ~env:[||] (Filename.concat p "git.exe")
+    in
+    let gits =
+      OpamStd.Env.get "PATH"
+      |> OpamStd.Sys.split_path_variable
+      |> OpamStd.List.filter_map (fun p ->
+          match contains_git p with
+          | Some git ->
+            Some (git, OpamSystem.bin_contains_bash p)
+          | None -> None)
+    in
+    let get_gitbin ?gitbin () =
+      let bin =
+        match gitbin with
+        | Some _ -> gitbin
+        | None ->
+          OpamConsole.read "Enter winget or Git for Windows binary path (\\bin):"
```

Comment:
I'm not sure what the "winget" bit means here? Is "Please enter the path containing git.exe (e.g. C:\Program Files\Git\cmd)" closer?

Reply:
```suggestion
          OpamConsole.read "Please enter the path containing git.exe (e.g. C:\Program Files\Git\cmd):"
```